utils/util_test.go

```diff
@@ -27,3 +28,35 @@ func TestBitsBytesConvert(t *testing.T) {
 		}
 	}
 }
+
+func TestIntToBytes(t *testing.T) {
+	numInt := 42
+	b := IntToBytes(numInt)
+	if int(binary.LittleEndian.Uint32(b)) != numInt {
+		t.Fatal("Conversion to bytes looks wrong!")
+	}
+	// TODO It is possible to call `IntToBytes` with a signed int < 0
+	// isn't that problematic?
```

Comment:
I think to let it `panic` or only use `UIntToBytes` would make sense. As far as I can see `node.level` can only grow (or never is a value subtracted from level). Maybe we could even refactor `level` to be unsigned?